go/vt/vttablet/tabletserver/vstreamer/vstreamer.go

```diff
@@ -342,17 +342,33 @@ func (vs *vstreamer) parseEvent(ev mysql.BinlogEvent) ([]*binlogdatapb.VEvent, e
 			vs.plans[id] = nil
 			return nil, nil
 		}
+
+		tableName := tm.Name
+		var cols []schema.TableColumn
 		st := vs.se.GetTable(sqlparser.NewTableIdent(tm.Name))
 		if st == nil {
-			return nil, fmt.Errorf("unknown table %v in schema", tm.Name)
-		}
-		if len(st.Columns) < len(tm.Types) {
-			return nil, fmt.Errorf("cannot determine table columns for %s: event has %d columns, current schema has %d: %#v", tm.Name, len(tm.Types), len(st.Columns), ev)
+			if vs.filter.BestEffortNameInFieldEvent == false {
+				// throw error. Column name might be need in filter.
```

Comment:
will remove this comment.

Reply:
`if !vs.filter.BestEffortNameInFieldEvent {`